### PR TITLE
Update medical_license_recognizer.py

### PR DESCRIPTION
**Pull request by andygonzalez2010**
Fix the bug in calculating the checksum and improve the regex

## Change Description

improved the regex based on: https://www.meditec.com/blog/dea-numbers-what-do-they-mean
fixed the checksum calculation to match the description in the url

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

File: presidio-analyzer/tests/test_certificate_number_recognizer.py [hunk 1]
```diff
@@ -18,10 +18,10 @@ def entities():
     "text, expected_len, expected_scores, expected_res",
     [
         # fmt: off
-        ("AS1111111 D91111111", 2, (), ((0, 9),(10,19),),),
-        ("AD2222222", 1, (), ((0, 9),),),
-        ("my certificate number is: AD2222222", 1, (), ((26,35),),),
-        ("The DEA number is  A61111111", 0, (), (),),
+        ("GL0285191 EU4488929", 2, (), ((0, 9),(10,19),),),
```
Comment (omri374) [inline presidio-analyzer/tests/test_certificate_number_recognizer.py:21]: Can we add those cases in addition and not instead? Is there a different reason for replacing?

Comment (andygonzalez2010) [reply presidio-analyzer/tests/test_certificate_number_recognizer.py:21]: the checksum calculation was broken, so the test was not actually valid licenses, I replaced them with valid ones according to https://www.meditec.com/blog/dea-numbers-what-do-they-mean

Comment (omri374) [reply presidio-analyzer/tests/test_certificate_number_recognizer.py:21]: Thanks. Looks good

**Review by omri374 — APPROVED** [review]
Thanks!

**Review by omri374 — DISMISSED** [review]

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/medical_license_recognizer.py [hunk 2]
```diff
@@ -21,8 +21,9 @@ class MedicalLicenseRecognizer(PatternRecognizer):
     PATTERNS = [
         Pattern(
             "USA DEA Certificate Number (weak)",
-            r"[a-zA-Z]{2}\d{7}|[a-zA-Z]{1}9\d{7}",
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/medical_license_recognizer.py:24]: Are the two new regex in addition or instead this weak one? Could you please put a link to the source in a comment on this file?

Comment (omri374) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/medical_license_recognizer.py:24]: The one from the other comment (https://www.meditec.com/blog/dea-numbers-what-do-they-mean) is good

**Review by omri374 — APPROVED** [review]
LGTM, thanks!

**Review by SharonHart — APPROVED** [review]

**Pull request merged**